src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,197 @@
+            throw new RuntimeException("Couldn't get a list of cars." + e);
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateStatement = connection.prepareStatement(query)) {
+            updateStatement.setString(1, car.getModel());
+            updateStatement.setLong(2, car.getManufacturer().getId());
+            updateStatement.setLong(3, car.getId());
+
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't update Car" + car, e);
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT c.id, model, manufacturers_id, name, country "
+                + "FROM cars_drivers cd JOIN cars c ON c.id = cd.car_id "
+                + "JOIN manufacturers m ON m.id = c.manufacturers_id "
+                + "WHERE c.is_deleted = FALSE AND cd.driver_id = ?";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Car car = getCar(resultSet);
+                cars.add(car);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Couldn't get a list of all car by driver's id'" + driverId, e);
+        }
+        cars.forEach(car -> car.setDrivers(getDriversForCar(car.getId())));
+        return cars;
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setModel(resultSet.getString("model"));
+        car.setId(resultSet.getLong("id"));
+        Manufacturer manufacturer = new Manufacturer();
+        manufacturer.setId(resultSet.getLong("id"));
+        manufacturer.setName(resultSet.getString("name"));
+        manufacturer.setCountry(resultSet.getString("country"));
+        car.setManufacturer(manufacturer);
+        return car;
+    }
+
+    private Driver parseDriver(ResultSet resultSet) throws SQLException {
```

Comment:
Method names have to be consistent.  getCar and parseDriver do the same work for Car and Driver. You should choose 'get' or 'parse' for both.